### PR TITLE
More events

### PR DESCRIPTION
**Pull request by urothis**

### CODE REVIEW

File: src/main/NWNX/API/Events/BarterEvents.cs [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    [NWNXEvent("NWNX_ON_BARTER_END_BEFORE")]
+    public class OnBarterEndBefore : EventSkippable<OnBarterEndBefore>
+    {
+      public NwPlayer Player { get; private set; }
+      public NwPlayer Target { get; private set; }
+      public bool Completed { get; private set; }
+      public int InitiatorItemCount { get; private set; }
+      public int TargetItemCount { get; private set; }
+      public NwObject[] PlayerItems { get; private set; }
+      public NwObject[] TargetItems { get; private set; }
```
Comment (Jorteck) [inline src/main/NWNX/API/Events/BarterEvents.cs:40]: These are not set.

File: src/main/NWNX/API/Events/AssociateEvents.cs [hunk 2]
```diff
@@ -0,0 +1,63 @@
+using NWN.API;
+using NWN.API.Events;
+using NWN.Core;
+using NWN.Core.NWNX;
+
+namespace NWNX.API.Events {
```
Comment (Jorteck) [inline src/main/NWNX/API/Events/AssociateEvents.cs:6]: New line for braces.

File: src/main/NWNX/API/Events/CalendarEvents.cs [hunk 3]
```diff
@@ -0,0 +1,86 @@
+using NWN.API;
+using NWN.API.Events;
+using NWN.Core;
+using NWN.Core.NWNX;
+
+namespace NWNX.API.Events {
+  public class CalendarEvents {
```
Comment (Jorteck) [inline src/main/NWNX/API/Events/CalendarEvents.cs:7]: New line for braces.

File: src/main/NWNX/API/Events/BarterEvents.cs [hunk 4]
```diff
@@ -0,0 +1,60 @@
+using NWN.API;
+using NWN.Core;
+using NWN.Core.NWNX;
+
+namespace NWNX.API.Events {
+  public class BarterEvents {
```
Comment (Jorteck) [inline src/main/NWNX/API/Events/BarterEvents.cs:6]: New line for braces.

File: src/main/NWNX/API/Events/BarterEvents.cs [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+    [NWNXEvent("NWNX_ON_BARTER_START_BEFORE")]
+    public class OnBarterStartBefore : EventSkippable<OnBarterStartBefore>
+    {
+      public NwPlayer Player { get; private set; }
+      public NwPlayer Target { get; private set; }
+
+      protected override void PrepareEvent(NwObject objSelf) {
```
Comment (Jorteck) [inline src/main/NWNX/API/Events/BarterEvents.cs:13]: New line for braces.

File: src/main/NWNX/API/Events/BarterEvents.cs [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+      }
+    }
+    
+    [NWNXEvent("NWNX_ON_BARTER_START_AFTER")]
+    public class OnBarterStartAfter : EventSkippable<OnBarterStartAfter>
+    {
+      public NwPlayer Player { get; private set; }
+      public NwPlayer Target { get; private set; }
+
+      protected override void PrepareEvent(NwObject objSelf) {
```
Comment (Jorteck) [inline src/main/NWNX/API/Events/BarterEvents.cs:25]: New line for braces.

File: src/main/NWNX/API/Events/BarterEvents.cs [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+    {
+      public NwPlayer Player { get; private set; }
+      public NwPlayer Target { get; private set; }
+      public bool Completed { get; private set; }
+      public int InitiatorItemCount { get; private set; }
+      public int TargetItemCount { get; private set; }
+      public NwObject[] PlayerItems { get; private set; }
+      public NwObject[] TargetItems { get; private set; }
+      
+      protected override void PrepareEvent(NwObject objSelf) {
```
Comment (Jorteck) [inline src/main/NWNX/API/Events/BarterEvents.cs:42]: New line for braces.

File: src/main/NWNX/API/Events/BarterEvents.cs [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+      }
+    }
+    
+    [NWNXEvent("NWNX_ON_BARTER_END_AFTER")]
+    public class OnBarterEndAfter : EventSkippable<OnBarterEndAfter>
+    {
+      public NwPlayer Player { get; private set; }
+      public NwPlayer Target { get; private set; }
+
+      protected override void PrepareEvent(NwObject objSelf) {
```
Comment (Jorteck) [inline src/main/NWNX/API/Events/BarterEvents.cs:54]: New line for braces.

File: src/main/NWNX/API/Events/ClientEvents.cs [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+    [NWNXEvent("NWNX_ON_CLIENT_CONNECT_BEFORE")]
+    public class OnClientConnectBefore : EventSkippable<OnClientConnectBefore>
+    {
+      public NwModule Module { get; private set; }
+      public string Username { get; private set; }
+      public string CDKey { get; private set; }
+      public bool DM { get; private set; }
+      public string IP { get; private set; }
+
+      protected override void PrepareEvent(NwObject objSelf) {
```
Comment (Jorteck) [inline src/main/NWNX/API/Events/ClientEvents.cs:35]: New line for braces.

File: src/main/NWNX/API/Events/ClientEvents.cs [hunk 10]
```diff
@@ -0,0 +1,62 @@
...
+    [NWNXEvent("NWNX_ON_CLIENT_CONNECT_AFTER")]
+    public class OnClientConnectAfter : EventSkippable<OnClientConnectAfter>
+    {
+      public NwModule Module { get; private set; }
+      public string Username { get; private set; }
+      public string CDKey { get; private set; }
+      public bool DM { get; private set; }
+      public string IP { get; private set; }
+
+      protected override void PrepareEvent(NwObject objSelf) {
```
Comment (Jorteck) [inline src/main/NWNX/API/Events/ClientEvents.cs:53]: New line for braces.

File: src/main/NWNX/API/Events/StoreEvents.cs [hunk 11]
```diff
@@ -0,0 +1,76 @@
+using NWN.API;
+using NWN.API.Events;
+using NWN.Core;
+using NWN.Core.NWNX;
+
+namespace NWNX.API.Events {
+  public class StoreEvents {
```
Comment (Jorteck) [inline src/main/NWNX/API/Events/StoreEvents.cs:7]: New line for braces.

File: src/main/NWNX/API/Events/StealthEvents.cs [hunk 12]
```diff
@@ -0,0 +1,42 @@
+using NWN.API;
+using NWN.API.Events;
+
+namespace NWNX.API.Events {
+  public class StealthEvents {
```
Comment (Jorteck) [inline src/main/NWNX/API/Events/StealthEvents.cs:5]: New line for braces.

**Review by Jorteck — CHANGES_REQUESTED** [review]
We should start documenting these events.

AreaEvents.OnEnter/AreaEvents.OnExit is a good example for what these classes should look like.

File: src/main/NWNX/API/Events/BarterEvents.cs (hunk 1 above)
Comment (urothis) [reply src/main/NWNX/API/Events/BarterEvents.cs:40]: Oops didn't mean to push that file yet. Was trying to determine the best way to create the collection.

Comment (urothis) [reply src/main/NWNX/API/Events/BarterEvents.cs]: I removed the event in question for the time being. Will add back later.

**Pull request merged**